link-check.js

```diff
@@ -3,13 +3,23 @@ const fs = require("fs");
 const path = require("path");
 const glob = require("glob");
 const chalk = require("chalk");
+const fetch = require("node-fetch");
+const pMap = require("p-map");
+const memoize = require("memoizee");
+const Entities = require("html-entities").AllHtmlEntities;
+
+const entities = new Entities();
 
 const base = `${__dirname}/public`;
 const files = glob.sync(`${base}/**/*.html`);
 const validLinks = files.map(f =>
   f.substr(base.length).replace(/index.html$/, "")
 );
 
+const checkLinkResolution = memoize(function checkLinkResolution(url) {
```

Comment:
This memoization properly restricts running fetches to only once per input url (i did actually test it :))